lib/policies/proxy/proxy.js

```diff
@@ -12,6 +12,11 @@ const createStrategy = (strategy, proxyOptions, endpointUrls) => {
 
 module.exports = function (params, config) {
   const serviceEndpointKey = params.serviceEndpoint;
+
+  if (!serviceEndpointKey) {
+    return (req, res) => res.sendStatus(404);
```

Comment:
how about 
502 Bad Gateway

   The 502 (Bad Gateway) status code indicates that the server, while
   acting as a gateway or proxy, received an invalid response from an
   inbound server it accessed while attempting to fulfill the request.

no response from the nonexisting server is an invalid response, right?

Reply:
Yes I think that makes sense. I'll change the returned status code

Reply:
Added a warning.

> Most likely this is some misconfiguration issue

I can't believe we're making this happen :trollface: